docs/Toy/toy/interpreter.py

```diff
@@ -0,0 +1,150 @@
+import operator
+from typing import Any
+from itertools import accumulate
+
+from dataclasses import dataclass
+
+from xdsl.dialects.builtin import TensorType, VectorType, ModuleOp
+from xdsl.interpreter import Interpreter, InterpreterFunctionTable
+from xdsl.utils.exceptions import InterpretationError
+
+from . import dialect as toy
+
+
+def run_toy_func(interpreter: Interpreter, name: str,
+                 args: tuple[Any, ...]) -> tuple[Any, ...]:
+    for op in interpreter.module.ops:
+        if isinstance(op, toy.FuncOp) and op.sym_name.data == name:
+            return run_func(interpreter, op, args)
+
+    raise InterpretationError(f'Could not find toy function with name: {name}')
+
+
+toy_ft = InterpreterFunctionTable()
+
+
```

Comment:
I am not sure whether I like the `InterpreterFunctionTable` as a global object. Could we not solve this by subclassing it? That should still work with (slightly adjusted) `.register` decorators, I think. What do you think?

Reply:
This is definitely the biggest topic to discuss here, I think. Subclassing might work, as long as we figure out a way for them to compose still. I think some of the design constraints will become clearer once the pdl interpreter written in this framework works.

Reply:
Went with a kind of subclassing with registration